web/modules/custom/parser/src/Service/XMLReader.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace Drupal\parser\Service;
+
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\node\Entity\Node;
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use SimpleXMLElement;
+
+/**
+ * Class XMLReader.
+ *
+ * Parse an XML file.
+ */
+class XMLReader {
+
+  protected $entityTypeManager;
+  protected $paragraphStorage;
+
+  /**
+   * XMLReader constructor.
+   *
+   * Needed for the EntityTypeManager dependency injection.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager interface.
+   *
+   * @throws \Exception
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+    try {
+      $this->paragraphStorage = $this->entityTypeManager->getStorage('paragraph');
+    }
+    catch (InvalidPluginDefinitionException $ipde) {
+      throw new \Exception('Exception on location_telephone paragraph,  ' . $ipde->getMessage());
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Exception
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Parses xml file with php function SimpleXML.
+   *
+   * @throws \Exception
+   */
+  public function parse($xmlFile) {
+    if (!is_file($xmlFile)) {
+      throw new \RuntimeException(sprintf('File "%s" does not exist.', $xmlFile));
+    }
+    if (!is_readable($xmlFile)) {
+      throw new \RuntimeException(sprintf('File "%s" cannot be read.', $xmlFile));
+    }
+    // Get all telephone paragraphs.
+    $phone_paragraphs = $this->paragraphStorage
+      ->loadByProperties(['type' => 'location_telephone']);
+    $phones = [];
+    foreach ($phone_paragraphs as $phone) {
+      $phones[$phone->id()] = $phone;
+    }
+    // Get all locations entities.
+    try {
+      $locations = $this->entityTypeManager
+        ->getStorage('node')
+        ->loadByProperties(['type' => 'location']);
```

Comment:
include field_location_cnsl_id here, so you get only that node

Reply:
done